lib/second_factor/actions/discourse_connect_provider.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+module SecondFactor::Actions
+  class DiscourseConnectProvider < Base
+    def skip_second_factor_auth?(params)
+      sso = get_sso(payload(params))
+      !current_user || sso.logout || !sso.require_2fa || @opts[:confirmed_2fa_during_login]
+    end
+
+    def second_factor_auth_skipped!(params)
+      sso = get_sso(payload(params))
+      return { logout: true, return_sso_url: sso.return_sso_url } if sso.logout
+      return { no_current_user: true } if !current_user
```

Comment:
minor: Moving this up may prevent the need to parse.

Reply:
We actually need to check for logout first because if there's no `current_user` and we get a logout SSO request, we want to return the user back to the SSO site. If we checked for `current_user` first, then the user would be prompted to login only for us to log them out again right away. Let me verify we have a test for this behavior...

Reply:
Yes, 2 tests: `spec/lib/second_factor/actions/discourse_connect_provider_spec.rb:131` and `spec/requests/session_controller_spec.rb:1317`.